test/Microsoft.Sbom.Targets.E2E.Tests/GenerateSbomE2ETests.cs

```diff
@@ -0,0 +1,346 @@
+    public void SetupLocator()
+    {
+        if (MSBuildLocator.CanRegister)
+        {
+            MSBuildLocator.RegisterDefaults();
+        }
+    }
+
+    [TestCleanup]
+    public void CleanOutputFolders()
+    {
+        var binDir = Path.Combine(projectDirectory, "bin");
+        var objDir = Path.Combine(projectDirectory, "obj");
+
+        try
+        {
+            if (Directory.Exists(binDir))
+            {
+                Directory.Delete(binDir, true);
+            }
+
+            if (Directory.Exists(objDir))
+            {
+                Directory.Delete(objDir, true);
+            }
+
+            ProjectCollection.GlobalProjectCollection.UnloadAllProjects();
+        }
+        catch (Exception ex)
+        {
+            Assert.Fail($"Failed to cleanup output directories. {ex}");
+        }
+    }
+
+    private Project SetupSampleProject()
+    {
+        // Create a Project object for ProjectSample1
+        var projectFile = Path.Combine(projectDirectory, "ProjectSample1.csproj");
+        var sampleProject = new Project(projectFile);
+
+        // Get all the expected default properties
+        SetDefaultProperties(sampleProject);
+
+        // Set the TargetFrameworks property to empty. By default, it sets this property to net6.0 and net8.0, which fails for net8.0 builds.
+        sampleProject.SetProperty("TargetFrameworks", string.Empty);
+
+        // Set the paths to the sbom-tool CLI tool and Microsoft.Sbom.Targets.dll
+        sampleProject.SetProperty("SbomToolBinaryOutputPath", sbomToolPath);
+        sampleProject.SetProperty("GenerateSbomTaskAssemblyFilePath", generateSbomTaskPath);
+
+        return sampleProject;
+    }
+
+    private void SetDefaultProperties(Project sampleProject)
+    {
+        expectedPackageName = sampleProject.GetPropertyValue("PackageId");
+        expectedVersion = sampleProject.GetPropertyValue("Version");
+        assemblyName = sampleProject.GetPropertyValue("AssemblyName");
+        configuration = sampleProject.GetPropertyValue("Configuration");
+
+        if (string.IsNullOrEmpty(expectedPackageName))
+        {
+            expectedPackageName = assemblyName;
+        }
+
+        if (string.IsNullOrEmpty(expectedPackageName))
+        {
+            expectedVersion = "1.0.0";
+        }
+    }
+
+    private void RestoreBuildPack(Project sampleProject)
+    {
+        var logger = new ConsoleLogger();
+
+        // Restore the project to create project.assets.json file
+        var restore = sampleProject.Build("Restore", new[] { logger });
+        Assert.IsTrue(restore, "Failed to restore the project");
+
+        // Next, build the project
+        var build = sampleProject.Build(logger);
+        Assert.IsTrue(build, "Failed to build the project");
+
+        // Finally, pack the project
+        var pack = sampleProject.Build("Pack", new[] { logger });
+        Assert.IsTrue(pack, "Failed to pack the project");
+    }
+
+    private void ExtractPackage()
```

Comment:
As mentioned in another comment, NuGet package extraction expects that the zip file's central directory uses `/`, never `\` for directory separators.  Therefore, I think it would be "better" if these tests do not extract to disk, but instead use `ZipArchive` to read the listing and contents of files, and importantly, validate that all path separators are always `/`, even when run on Windows.

Reply:
Thank you @zivkan. I have updated the tests in fe6c158099f9df496a4ffd62e1cdaa8a800da461